### PR TITLE
Servers Page Refactoring

### PR DESCRIPTION
**Pull request by caseyswilliams**
- Moved some stuff around, did some better mapping/distinction between Hubs/Servers/Quick Play/Match
- Corrected filtering rules for Hubs/Servers/Quick Play
- Added Servers and Quick Play to Servers page

### CODE REVIEW

File: UT4MasterServer.Web/src/pages/Servers/data/get-servers-test-response.ts [hunk 1]
```diff
@@ -0,0 +1,2033 @@
+// Sample response from matchMakingRequest for testing
+export const getServersTestResponse = [
```
Comment (timiimit) [inline UT4MasterServer.Web/src/pages/Servers/data/get-servers-test-response.ts:2]: you can use example from from before epic shut down servers. it's located in `OldReferenceCode/HubMenuResponse.json`. It's longer by 1000 lines.

File: UT4MasterServer.Web/src/pages/Servers/enums/game-mode.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// This isn't a complete list, but it is only used in quick play for now and I think these are all the possible values for quick play
+export enum GameMode {
+  duel = '/Script/UnrealTournament.UTDuelGameMode',
+  deathmatch = '/Script/UnrealTournament.UTDMGameMode',
+  ctf = '/Script/UnrealTournament.UTCTFGameMode',
+  blitz = '/Script/UnrealTournament.UTFlagRunGame',
+  empty = 'EMPTY',
+  hub = '/Script/UnrealTournament.UTLobbyGameMode'
```
Comment (timiimit) [inline UT4MasterServer.Web/src/pages/Servers/enums/game-mode.ts:8]: Add all stock game modes:
siege = `/Script/UnrealTournament.UTFlagRunPvEGame`
team_deathmatch = `/Script/UnrealTournament.UTTeamDMGameMode`
showdown= `/Script/UnrealTournament.UTShowdownGame`

And commonly played custom game modes which will be possible to play in quickplay:
elimination = `/Game/ALTS/Elimination_113.Elimination_113_C`
bunnytrack = `/Game/BT/Gamemode/BT_Game.BT_Game_C` // not confirmed, may be wrong

File: UT4MasterServer.Web/src/pages/Servers/enums/match-state.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// I don't think this is a complete list, but I couldn't locate one. Can be easily added to, and will default to the attribute value which is what it used to display.
+export enum MatchState {
+  waiting = 'WaitingToStart',
+  inProgress = 'InProgress',
+  intermission = 'MatchIntermission',
+  playerIntro = 'PlayerIntro'
```
Comment (timiimit) [inline UT4MasterServer.Web/src/pages/Servers/enums/match-state.ts:6]: from source code:
```cpp
	extern UNREALTOURNAMENT_API const FName PlayerIntro;					// Playing the player intro in the match summary window
	extern UNREALTOURNAMENT_API const FName CountdownToBegin;				// We are entering this map, actors are not yet ticking
	extern UNREALTOURNAMENT_API const FName MatchEnteringOvertime;			// The game is entering overtime
	extern UNREALTOURNAMENT_API const FName MatchIsInOvertime;				// The game is in overtime
	extern UNREALTOURNAMENT_API const FName MapVoteHappening;				// The game is in mapvote stage
	extern UNREALTOURNAMENT_API const FName MatchIntermission;				// The game is in a round intermission
	extern UNREALTOURNAMENT_API const FName MatchExitingIntermission;		// Exiting Halftime
	extern UNREALTOURNAMENT_API const FName MatchRankedAbandon;				// Exiting Halftime
	extern UNREALTOURNAMENT_API const FName WaitingTravel;					// The client is awaying travel to the next map
```

File: UT4MasterServer.Web/src/pages/Servers/data/get-servers-test-response.ts (hunk 1 above)
Comment (caseyswilliams) [reply UT4MasterServer.Web/src/pages/Servers/data/get-servers-test-response.ts:2]: That's good to know that it is there, but I'd rather have the response from the new master server than the old.

File: UT4MasterServer.Web/src/pages/Servers/enums/game-mode.ts (hunk 2 above)
Comment (caseyswilliams) [reply UT4MasterServer.Web/src/pages/Servers/enums/game-mode.ts:8]: Added

File: UT4MasterServer.Web/src/pages/Servers/enums/match-state.ts (hunk 3 above)
Comment (caseyswilliams) [reply UT4MasterServer.Web/src/pages/Servers/enums/match-state.ts:6]: Added

**Review by timiimit — APPROVED** [review]

**Pull request merged**